src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesOwaspDependencyCheck.java

```diff
@@ -40,82 +66,250 @@ public UsesOwaspDependencyCheck(GitHubDataFetcher fetcher) {
+      return values;
+    }
+
+    checkGradle(repository, values);
+    if (found(values)) {
+      return values;
+    }
+
+    values.update(OWASP_DEPENDENCY_CHECK_USAGE.value(NOT_USED));
+    values.update(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD.notSpecifiedValue());
+
+    return values;
+  }
+
+  /**
+   * Checks if a value set contains the features for OWASP Dependency Check.
+   *
+   * @param values The value set to be checked.
+   * @return True if the value set contains the feature, false otherwise.
+   */
+  private static boolean found(ValueSet values) {
+    return values.has(OWASP_DEPENDENCY_CHECK_USAGE)
+        && values.has(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD);
   }
 
   /**
-   * Checks if a project uses OWASP Dependency Check Maven plugin.
+   * Checks if a project uses OWASP Dependency Check Maven plugin,
+   * and updates a value set if the plugin is found.
    *
    * @param repository The project's repository.
-   * @return True if the project uses the plugin, false otherwise.
+   * @param values A {@link ValueSet} to be updated.
+   * @throws IOException If something goes wrong.
    */
-  private boolean checkMaven(LocalRepository repository) throws IOException {
+  private static void checkMaven(LocalRepository repository, ValueSet values) throws IOException {
     Optional<InputStream> content = repository.read("pom.xml");
 
-    if (!content.isPresent()) {
-      return false;
-    }
+    if (content.isPresent()) {
+      try (InputStream is = content.get()) {
+        Model model = readModel(is);
+
+        Visitor visitor = browse(model, withVisitor());
+
+        OwaspDependencyCheckUsageValue usage = OWASP_DEPENDENCY_CHECK_USAGE.value(visitor.usage());
+
+        OwaspDependencyCheckCvssThresholdValue threshold = visitor.threshold()
+            .map(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD::value)
+            .orElse(OWASP_DEPENDENCY_CHECK_FAIL_CVSS_THRESHOLD.notSpecifiedValue());
 
-    try (InputStream is = content.get()) {
-      Model model = readModel(is);
-      return browse(model, withVisitor()).result;
+        values.update(usage, threshold);
+      }
     }
   }
 
   /**
-   * Checks if a project uses OWASP Dependency Check Gradle plugin.
+   * Checks if a project uses OWASP Dependency Check Gradle plugin,
+   * and updates a value set with corresponding feature values.
    *
    * @param repository The project's repository.
-   * @return True if the project uses the plugin, false otherwise.
+   * @param values A {@link ValueSet} to be updated.
    */
-  private boolean checkGradle(LocalRepository repository) throws IOException {
-    Optional<InputStream> content = repository.read("build.gradle");
+  private static void checkGradle(LocalRepository repository, ValueSet values) throws IOException {
+    for (Path gradleFile : repository.files(path -> path.getFileName().endsWith(".gradle"))) {
+      Optional<List<String>> something = repository.readLinesOf(gradleFile);
```

Comment:
Are we closing the stream here when reading lines?

Reply:
`readLinesOf()` closes the stream.